src/app/modules/upload-status/users-csv/users-csv.component.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
+import { MatPaginator, MatTableDataSource, PageEvent, MatDialog } from '@angular/material';
+import { BulkuploadService } from '../../admin-core';
+import { SelectionModel } from '@angular/cdk/collections';
+import { fromEvent, of, Observable } from 'rxjs';
+import { distinctUntilChanged, map, debounceTime, filter } from 'rxjs/operators';
+import { FormControl } from '@angular/forms';
+
+
+@Component({
+  selector: 'app-users-csv',
+  templateUrl: './users-csv.component.html',
+  styleUrls: ['./users-csv.component.scss']
+})
+export class UsersCsvComponent implements OnInit {
+  toppings = new FormControl();
+  toppingList: string[] = ['Extra cheese', 'Mushroom', 'Onion', 'Pepperoni', 'Sausage', 'Tomato'];
+  dataSource: MatTableDataSource<any>;
+  displayedColumns: any = [];
+  selection = new SelectionModel(true, []);
+  listing: boolean = false;
+  recordCount: any;
+  initialStatus: any = '';
+  initialType: any = '';
+  status: any = '';
+  columns: any;
+  statusList = [
+    {
+      label: 'All',
+      value: ''
+    }, {
+      label: 'Success',
+      value: 'Success'
+    }, {
+      label: 'Pending',
+      value: 'pending'
+    }, {
+      label: 'Failed',
+      value: 'Failed'
+    }];
+
+  requestType = [{
+    label: 'All',
+    value: ''
+  }, {
+    label: 'user Creation',
+    value: 'user Creation'
+  },]
+  queryParams = {
+    page: 1,
+    size: 10,
+  };
+  organisationListData: any;
+  @ViewChild('searchInput') searchInput: ElementRef;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  constructor(private bulkuploadService: BulkuploadService) { }
+
+  ngOnInit() {
+
+    fromEvent(this.searchInput.nativeElement, 'keyup').pipe(
+      // get value
+      map((event: any) => {
+        return event.target.value;
+      })
+      // if character length greater then 2
+      , filter(res => res.length > 2 || res.length == 0)
+      // Time in milliseconds between key events
+      , debounceTime(1000)
```

Comment:
Why you are using this debounce here, we are having the directive now

Reply:
Removed